substrate/frame/support/test/tests/runtime_legacy_ordering.rs

```diff
@@ -0,0 +1,993 @@
+			pub _config: sp_std::marker::PhantomData<T>,
+		}
+
+		#[pallet::genesis_build]
+		impl<T: Config> BuildGenesisConfig for GenesisConfig<T> {
+			fn build(&self) {}
+		}
+
+		#[pallet::validate_unsigned]
+		impl<T: Config> ValidateUnsigned for Pallet<T> {
+			type Call = Call<T>;
+			fn validate_unsigned(
+				_source: TransactionSource,
+				_call: &Self::Call,
+			) -> TransactionValidity {
+				Err(TransactionValidityError::Invalid(InvalidTransaction::Call))
+			}
+		}
+	}
+}
+
+#[frame_support::pallet(dev_mode)]
+pub mod module3 {
+	use super::*;
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		pub fn fail(_origin: OriginFor<T>) -> DispatchResult {
+			Err(Error::<T>::Something.into())
+		}
+		pub fn aux_1(_origin: OriginFor<T>, #[pallet::compact] _data: u32) -> DispatchResult {
+			unreachable!()
+		}
+		pub fn aux_2(
+			_origin: OriginFor<T>,
+			_data: i32,
+			#[pallet::compact] _data2: u32,
+		) -> DispatchResult {
+			unreachable!()
+		}
+		#[pallet::weight(0)]
+		pub fn aux_3(_origin: OriginFor<T>, _data: i32, _data2: String) -> DispatchResult {
+			unreachable!()
+		}
+		#[pallet::weight(3)]
+		pub fn aux_4(_origin: OriginFor<T>) -> DispatchResult {
+			unreachable!()
+		}
+		#[pallet::weight((5, DispatchClass::Operational))]
+		pub fn operational(_origin: OriginFor<T>) -> DispatchResult {
+			unreachable!()
+		}
+	}
+
+	#[pallet::origin]
+	#[derive(Clone, PartialEq, Eq, RuntimeDebug, Encode, Decode, MaxEncodedLen, TypeInfo)]
+	pub struct Origin<T>(pub PhantomData<T>);
+
+	#[pallet::event]
+	pub enum Event<T> {
+		A,
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		Something,
+	}
+
+	#[pallet::genesis_config]
+	#[derive(frame_support::DefaultNoBound)]
+	pub struct GenesisConfig<T: Config> {
+		#[serde(skip)]
+		pub _config: sp_std::marker::PhantomData<T>,
+	}
+
+	#[pallet::genesis_build]
+	impl<T: Config> BuildGenesisConfig for GenesisConfig<T> {
+		fn build(&self) {}
+	}
+
+	#[pallet::storage]
+	pub type Storage<T> = StorageValue<_, u32>;
+
+	#[pallet::validate_unsigned]
+	impl<T: Config> ValidateUnsigned for Pallet<T> {
+		type Call = Call<T>;
+		fn validate_unsigned(
+			_source: TransactionSource,
+			_call: &Self::Call,
+		) -> TransactionValidity {
+			Err(TransactionValidityError::Invalid(InvalidTransaction::Call))
+		}
+	}
+}
+
+pub type BlockNumber = u64;
+pub type Signature = sr25519::Signature;
+pub type AccountId = <Signature as Verify>::Signer;
+pub type Header = generic::Header<BlockNumber, BlakeTwo256>;
+pub type UncheckedExtrinsic = generic::UncheckedExtrinsic<u32, RuntimeCall, Signature, ()>;
+pub type Block = generic::Block<Header, UncheckedExtrinsic>;
+
+#[frame_support::runtime(legacy_ordering)]
+mod runtime {
```

Comment:
Before using this in Rococo/Kusama we could add a test that iterates through all tests and writes their index to a JSON file.  
Then once the V2 syntax is used we can assert that nothing changed.

Reply:
Could you explain this a bit more? You mean iterate through all tests present in the repo and store their indices? This file is actually using the exact same tests as with the older `construct_runtime` syntax, thus already ensuring that `legacy_ordering` with the new runtime macro doesn't change anything?

Reply:
Thinking about this a bit more: we can just generate the runtime metadata before using V2 CR and then afterwards.  
Their diff should be empty if we use it correctly. Was just thinking about how we can ensure that we dont accidentally mess up Kusama, but should be fine.